File: python/longest_consecutive_squence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        numsSet = set(nums)
        longest = 0
        for num in nums:
            if num not in numsSet:
                continue
            # Grow upward
            l, k = 0, num+1
            while k in numsSet:
                numsSet.remove(k)
                k += 1
                l += 1
            # Grow downward
            k = num-1
            while k in numsSet:
                numsSet.remove(k)
                k -= 1
                l += 1

            longest = max(longest, l+1)

        return longest


if __name__ == '__main__':
    s = Solution()
    print s.longestConsecutive([1,2,3,3, 100, 200])

File: cpp/longest_consecutive_squence.cpp
#include <iostream>
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
    	// Set up set
    	unordered_set<int> numsSet;
        for (int i = 0; i < nums.size(); i++) 
        	numsSet.insert(nums[i]);
        // longest length
        int longest = 0;
        //
        for (int i = 0; i < nums.size(); i++) {
        	if (numsSet.find(nums[i]) == numsSet.end())
        		continue;
        	// Tmp length
        	int l = 0;
        	// Grow upward
        	int k = nums[i]+1;
        	while (numsSet.find(k) != numsSet.end()) {
        		numsSet.erase(k);
        		k++;
        		l++;
        	}
        	// Grow downward
        	k = nums[i] -1;
        	while (numsSet.find(k) != numsSet.end()) {
        		numsSet.erase(k);
        		k--;
        		l++;
        	}
        	// Refresh longest length
        	longest = max(longest, l+1);
        }

        return longest;
    }
};

int main(int argc, char const *argv[])
{
	Solution s;
	vector<int> nums = {1,2,3,3,100,200};
    cout<<"\n\n\n"<<s.longestConsecutive(nums);
	return 0;
}